### DISCUSSION TITLE
Improve documentation about WOODPECKER_HOST wrt. port number

### DISCUSSION THREAD

**Discussion by svdHero** [woodpecker-ci/woodpecker#2275] [General]
### Clear and concise description of the problem

In the [setup documentation](https://woodpecker-ci.org/docs/administration/setup#docker-compose) it reads:

> Woodpecker needs to know its own address. You must therefore provide the public address of it in `<scheme>://<hostname>` format.
> [...]
> `WOODPECKER_HOST=${WOODPECKER_HOST}`

It is not clear if `<scheme>://<hostname>` must contain the http port, e.g., the default port 8000 from the compose file. The [reference](https://woodpecker-ci.org/docs/administration/server-config#woodpecker_host) is not clear about this either.

### Suggested solution

Improve the wording in order to clarify if the port number has to be part of the environment variable `WOODPECKER_HOST`.

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by svdHero** [comment]
@6543 I am happy to work on this if you answer the question. I would assume that one has to add the port to the hostname. Could you confirm that?

**Comment by anbraten** [comment]
I has to contain the url the way you would enter it in the browser to access it. So in case of http you need to specifiy the port if its not `80` and in case of https if its not `443`.

**Comment by svdHero** [comment]
Thanks. That's what I thought. I will add a sentence or two to the documentation and open a PR.

**Comment by smainz** [comment]
Keep in mind: `WOODPECKER_HOST` is used to create urls which refere to the Web-UI as you type it when you access it with your browser.

- You do not have to add the port number, if it is the standard port i.e. port 80 for http and 443 for https.
- If you are behind a reverse proxy, it is the `<scheme>://<host>:<port>`of the reverse proxy.

HTH

**Comment by svdHero** [comment]
> Keep in mind: `WOODPECKER_HOST` is used to create urls which refere to the Web-UI as you type it when you access it with your browser.

That is exactly the sentence that I was looking for and that is missing in the documentation. I shall go and add it. 😄
